microsoft/testsuites/core/vm_hot_resize.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features.resize import Resize
+from lisa.schema import NodeSpace
+
+
+@TestSuiteMetadata(
+    area="vm_hot_resize",
+    category="functional",
+    description="""
+    This test suite tests vm behavior upon resizing without shutting down
+    """,
+)
+class VmHotResize(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case resizes the node and checks if it has the expected capabilities
+        (memory size and core count) after the resize
+
+        Steps:
+        1. Resize vm
+        2. Check the node's core count and memory size against their expected values
+        """,
+        priority=1,
+    )
+    def verify_vm_hot_resize(self, node: Node) -> None:
+        resize = node.features[Resize]
+        expected_vm_capability = resize.resize()
+        self._verify_cores_and_memory(node, expected_vm_capability)
+
+    def _verify_cores_and_memory(
+        self, node: Node, expected_vm_capability: NodeSpace
+    ) -> None:
+        expected_core_count = expected_vm_capability.core_count
+        actual_core_count = node.capability.core_count
+        assert_that(actual_core_count).described_as(
+            "The VM resize succeeded but the amount of cores that the vm has is "
+            f"incorrect. Expected {expected_core_count} cores but actually had "
+            f"{actual_core_count} cores"
+        ).is_equal_to(expected_core_count)
+
+        expected_memory_size = expected_vm_capability.memory_mb
```

Comment:
It doesn't need to check memory size. It's tricky to get the accurate memory value in system.

Reply:
removed